components/core/src/clp_s/JsonParser.cpp

```diff
@@ -154,23 +139,23 @@ void JsonParser::parse_line(ondemand::value line, int32_t parent_node_id, std::s
                         = std::string(raw_json_token.substr(1, raw_json_token.rfind('"') - 1));
 
                 if (matches_timestamp) {
-                    node_id = m_schema_tree->add_node(
+                    node_id = m_archive_writer->add_node(
                             node_id_stack.top(),
                             NodeType::DATESTRING,
                             cur_key
                     );
                     uint64_t encoding_id{0};
                     epochtime_t timestamp
-                            = m_timestamp_dictionary
+                            = m_archive_writer
                                       ->ingest_entry(m_timestamp_key, node_id, value, encoding_id);
```

Comment:
Maybe `ingest_timestamp_entry` would be more clear now that we go through the archive writer instead of directly to the timestamp dictionary?

Reply:
Ditto for other variants of this method

Reply:
Right